### PR TITLE
allow `quality` argument to be passed to `save`

### PR DESCRIPTION
**Pull request by jamarju**

### CODE REVIEW

**Comment by Isotr0py** [comment]
Thanks for your contribution!

Note that the `quality` used in `jpegxl-rs` is the `FrameDistance` in fact, which ranges from `0 .. 25` and `lower = higher quality`. 

I'm afraid that if we just expose `quality` simply, this may cause a confusing situation. Because a JPEG-style quality factor usually ranges up to 100 and `higher = higher quality`.

File: pillow_jxl/JpegXLImagePlugin.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def _save(im, fp, filename, save_all=False):
 
     # default quality is 1
     lossless = info.get("lossless", False)
-    quality = 0 if lossless else 1
+    quality = 0 if lossless else info.get("quality", 1)
```
Comment (Isotr0py) [inline pillow_jxl/JpegXLImagePlugin.py:90]: Maybe we can add a mapping from JPEG-Style quality to `FrameDistance` before `jpegxl-rs` fix the mismatch bug?

**Review by Isotr0py — APPROVED** [review]

File: pillow_jxl/JpegXLImagePlugin.py (hunk 1 above)
Comment (jamarju) [reply pillow_jxl/JpegXLImagePlugin.py:90]: I think you are right and a further refinement could be to use this [JxlEncoderDistanceFromQuality](https://libjxl.readthedocs.io/en/latest/api_encoder.html#_CPPv429JxlEncoderDistanceFromQualityf) function which provides a quality mapping similar to that of jpeg.

**Comment by jamarju** [comment]
I added a mapping according to aforementioned function, take a look and let me know what you think. Might be even cleaner to call the native function from the rust side of the code?

**Comment by Isotr0py** [comment]
LGTM.

I will update the forked `jpegxl-rs` with [JxlEncoderDistanceFromQuality](https://libjxl.readthedocs.io/en/latest/api_encoder.html#_CPPv429JxlEncoderDistanceFromQualityf) and submit a PR to `jpegxl-rs` main repo for further refinement in the near future.

**Pull request merged**